internal/stringutil/htmlchunker.go

```
@@ -0,0 +1,421 @@
+		return nil, err
+	}
+
+	var boundaries []htmlBoundary
+
+	var extract func(*html.Node)
+	extract = func(n *html.Node) {
+		if n.Type == html.ElementNode {
+			tag := strings.ToLower(n.Data)
+
+			var content strings.Builder
+			html.Render(&content, n)
```

golangci-lint (2.12.2)

[error] Error return value of `html.Render` is not checked

(errcheck)